_posts/2017-08-18-gsoc_2017-t0day.markdown

```diff
@@ -13,7 +13,161 @@ documentation:
+The JSON file is formatted like this 
+
+```
+To a leaf node (malware)
+name:					// sha256 name
+final_score:			// Final_score
+indicators:				// the reason that two samples are related
+
+To a branch node
+name:					// branch_name
+final_score:			// the max score of the children
+indicators:         	// the number of children
+children: []        	// the children of this node
+```
+
+#### Visualization
+Visualization part is a website displaying the final relationships using d3.js. There are multiple views and tricks to improve the user experience. The page shows the relationships in a tree map.  
+[See here](http://120.77.40.25/d3/index.html)
+
+##### Multiple Views
+These three views can be used together.
+
+1. Threshold View:
+When we set a threshold score, those nodes whose scores are greater than the threshold are displayed.
+2. Artefact View:
+When we click the artefact’s name, the nodes with this feature are added to ( or deleted from ) the map.
+3. Quick View:
+This view shows a histogram that displays the number of nodes in different segments of scores.
+
+##### Tricks
+
+1. Collapsible:
+When we click on the branch node, the branch node’s leaf nodes are folded and also the branch node is filled with steelblue.
+2. Zoomable:
+We can change the map scale by mouse wheel and also translate the map by dragging.
+3. Tooltip:
+When we move the mouse to the leaf node, the tooltip shows the final score, the reason that two samples are related, and the degree of this relation.  
+When we move the mouse to the branch node, the tooltip shows the branch name, the number of children in this branch, and the maximum score in this branch.
+4. Visualization weight:
+In the whole map, the line weight is set according to the target node score. In each set, the nodes are sorted by the score.
+
+### Proud of things
+Before the project, although I know something about the big data ecosystem, the actual operation of code is hardly. So along with the way, I encountered a lot of problems. Through communication with mentors, reading books, most of the problems have been solved.  
+These are things that make me impressed:
+
+1. There will always be some strange analysis results generated by Holmes-Totem and Holmes-Totem-Dynamic. For example, the preprocessing code worked well when I used dataframe to show the results. However, a few days later, it failed when I save all the results into the file. I tried to find the wrong sample, but failed. There is a little while, I was incompetent and weak, thinking to skip this problem and to delete the wrong samples. But I did not give up, finally finally and finally, through the `try ... except` function, I find the sample, and modify the code successfully. In fact, only six wrong samples in the millions of results.
```

Comment:
these will always appear when dealing with security data. It makes you always stay on your toes! It's fun!